crates/nu-parser/tests/test_parser.rs

```diff
@@ -364,6 +324,208 @@ fn test_nothing_comparisson_neq() {
+        }
+
+        #[test]
+        pub fn parse_string_interpolation_escaped_backslash_before_parenthesis() {
+            let engine_state = EngineState::new();
+            let mut working_set = StateWorkingSet::new(&engine_state);
+
+            let (block, err) = parse(
+                &mut working_set,
+                None,
+                b"$\"hello \\\\(39 + 3)\"",
+                true,
+                &[],
+            );
+
+            assert!(err.is_none());
+
+            assert!(block.len() == 1);
+            let expressions = &block[0];
+
+            assert!(expressions.len() == 1);
+            let expr = &expressions[0].expr;
+
+            let subexprs: Vec<&Expr>;
+            match expr {
+                Expr::StringInterpolation(expressions) => {
+                    subexprs = expressions.iter().map(|e| &e.expr).collect();
+                }
+                _ => panic!("Expected an `Expr::StringInterpolation`"),
+            }
+
+            assert_eq!(subexprs.len(), 2);
+
+            assert_eq!(subexprs[0], &Expr::String("hello \\".to_string()));
+
+            assert!(matches!(subexprs[1], &Expr::FullCellPath(..)));
+        }
+
+        #[test]
+        pub fn parse_string_interpolation_backslash_count_reset_by_expression() {
+            let engine_state = EngineState::new();
+            let mut working_set = StateWorkingSet::new(&engine_state);
+
+            let (block, err) = parse(
+                &mut working_set,
+                None,
+                b"$\"\\(1 + 3)\\(7 - 5)\"",
+                true,
+                &[],
+            );
+
+            assert!(err.is_none());
+
+            assert!(block.len() == 1);
+            let expressions = &block[0];
+
+            assert!(expressions.len() == 1);
+            let expr = &expressions[0].expr;
+
+            let subexprs: Vec<&Expr>;
+            match expr {
+                Expr::StringInterpolation(expressions) => {
+                    subexprs = expressions.iter().map(|e| &e.expr).collect();
+                }
+                _ => panic!("Expected an `Expr::StringInterpolation`"),
+            }
+
+            assert_eq!(subexprs.len(), 1);
+            assert_eq!(subexprs[0], &Expr::String("(1 + 3)(7 - 5)".to_string()));
+        }
+
+        #[test]
+        pub fn parse_nested_expressions() {
+            let engine_state = EngineState::new();
+            let mut working_set = StateWorkingSet::new(&engine_state);
+
+            working_set.add_variable(
+                "foo".to_string().into_bytes(),
+                Span { start: 0, end: 0 },
+                nu_protocol::Type::CellPath,
+                false,
+            );
+
+            let (_block, err) = parse(
+                &mut working_set,
+                None,
+                br#"
+                $"(($foo))"
+                "#,
+                true,
+                &[],
+            );
```

Comment:
Is this our only directly named variable expansion unit test? (Sure that in the general nu tests there might be a tested usage.) Do we want to include a full cell path expression like in the scripts that failed for @fdncred (`$"Hello ($foo.bar)")

Reply:
I've added an additional test `parse_path_expression` (duplicating `parse_nested_expressions`) for the expression you suggested.